test/meson.build

```diff
@@ -0,0 +1,84 @@
+add_languages('cpp')
+
+gtest_sources = [
+  'gtest/src/gtest-all.cc',
+  'gtest/src/gtest_main.cc',
+]
+
+libgtest_includes = [
+  include_directories('gtest'),
+  include_directories('gtest/include'),
+]
+
+cpp_flags = [
+  '-DGTEST_HAS_PTHREAD=' + thread_dep.found().to_string('1', '0'),
+  '-DGTEST_LANG_CXX11=1',
+  '-DGTEST_USE_OWN_TR1_TUPLE=0',
+  '-DGTEST_HAS_TR1_TUPLE=1',
+]
```

Comment:
Please add these to the test_i965_drv_video target, too, since they are used in gtest headers included by both targets.